app/models/report.rb

```diff
@@ -1480,12 +1497,17 @@ def self.report_top_uploads(report)
     FROM uploads up
     JOIN users u
     ON u.id = up.user_id
-    WHERE up.created_at >= '#{report.start_date}' AND up.created_at <= '#{report.end_date}'
+    /*where*/
     ORDER BY up.filesize DESC
     LIMIT #{report.limit || 250}
     SQL
 
-    DB.query(sql).each do |row|
+    extension_filter = report.filter_values["file-extension"]
```

Comment:
@jjaffeux is this what you had in mind? Also, I used the SQL Builder below 👇 

I believe any new report that uses custom filters will need to add a single line per filter, I feel it is alright, but happy to hear your feedback 😃 

```ruby
builder.where("up.extension = :extension", extension: extension_filter) if extension_filter.present?
```